custom_components/dahua/__init__.py

```diff
@@ -355,6 +360,19 @@ def on_receive_vto_event(self, event: dict):
 
         # This is the event code, example: VideoMotion, CrossLineDetection, BackKeyLight, PhoneCallDetect, DoorStatus, etc
         code = self.translate_event_code(event)
+        
+        # OZZIII BEGIN  FIRE TAG READER
+        if code == "AccessControl":
+            card_id = event.get("Data", {}).get("CardNo", "")
+            if card_id:
+                card_id_md5 = hashlib.md5(card_id.encode()).hexdigest()
+                asyncio.run_coroutine_threadsafe(
+                    async_scan_tag(self.hass, card_id_md5, self.get_device_name()), self.hass.loop
+                ).result()
+                return
```

Comment:
Are you certain this return needs to be here?

Reply:
I am not the owner of the PR, but I confirm that the code works once you remove the unneeded `return`, as you very well pointed out.

Shall we wait for @ozziii to fix it, or do you want me to do a separate PR?